modules/local/check_duplicates.nf

```diff
@@ -1,32 +1,44 @@
 process CHECK_DUPLICATES {
     label 'process_low'
+    tag "${meta.id}"
 
     conda "conda-forge::pigz=2.6"
     container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
         'https://depot.galaxyproject.org/singularity/ubuntu:24.04' :
         'biocontainers/ubuntu:24.04' }"
 
     input:
-    path fnas
+    tuple val(meta), path(fna)
 
     output:
-    path "duplicate-contigs.txt" , emit: duplicate_list, optional: true
-    path "versions.yml"          , emit: versions
-
-    when:
-    task.ext.when == null || task.ext.when
+    path "duplicates.txt"            , emit: duplicates_file, optional: true
+    tuple val(meta), path(fna)       , emit: fna, optional: true
+    path "versions.yml"              , emit: versions
```

Comment:
```suggestion
    path "${prefix}.duplicate-contigs.txt"         , emit: duplicate_contig_names, optional: true
    tuple val(meta), path("${prefix}.fnas_with_duplicates.txt"), emit: fnas_with_duplicates, optional: true
    path "versions.yml"              , emit: versions
    
    when:
    task.ext.when == null || task.ext.when
```
I don't think they need to be optional. Did you make them optional to easier test for that in the surrounding code?
IMO, making the file names a channel of names rather than of a single file -- i.e. declaring it as `stdout`, see below -- would make things easier. In the surrounding code, you could then just check with `.empty()` and fail if it's not.
```
    stdout, emit fnas_with_duplicates
```